File: src/main/java/sawyern/cookiebot/repository/WeeklyCooldownRepository.java
package sawyern.cookiebot.repository;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;
import sawyern.cookiebot.models.entity.Account;
import sawyern.cookiebot.models.entity.WeeklyCooldown;

@Repository
public interface WeeklyCooldownRepository extends CrudRepository<WeeklyCooldown, String> {
    WeeklyCooldown findByAccount(Account account);
}

File: src/main/kotlin/sawyern/cookiebot/repository/WeeklyCooldownRepository.kt
package sawyern.cookiebot.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import sawyern.cookiebot.models.WeeklyCooldown

@Repository
interface WeeklyCooldownRepository: JpaRepository<WeeklyCooldown, String> {
    fun findByAccountDiscordId(discordId: String): WeeklyCooldown?
}